src/main/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParams.java

```diff
@@ -447,67 +449,65 @@ public String buildQueryParamString(Long forPage) {
         List<String> queries = new ArrayList<>();
 
         if (accountIds != null && !accountIds.isEmpty()) {
-            queries.add(GATEWAY_ACCOUNT_EXTERNAL_FIELD + "=" + String.join(",", accountIds));
+            queries.add(GATEWAY_ACCOUNT_EXTERNAL_FIELD + "=" + URLEncoder.encode(String.join(",", accountIds), StandardCharsets.UTF_8));
         }
         if (isNotBlank(fromDate)) {
-            queries.add(FROM_DATE_FIELD + "=" + fromDate);
+            queries.add(FROM_DATE_FIELD + "=" + URLEncoder.encode(fromDate, StandardCharsets.UTF_8));
         }
         if (isNotBlank(toDate)) {
-            queries.add(TO_DATE_FIELD + "=" + toDate);
+            queries.add(TO_DATE_FIELD + "=" + URLEncoder.encode(toDate, StandardCharsets.UTF_8));
         }
-
         if (isNotBlank(email)) {
-            queries.add(EMAIL_FIELD + "=" + email);
+            queries.add(EMAIL_FIELD + "=" + URLEncoder.encode(email, StandardCharsets.UTF_8));
         }
         if (isNotBlank(reference)) {
-            queries.add(REFERENCE_FIELD + "=" + reference);
+            queries.add(REFERENCE_FIELD + "=" + URLEncoder.encode(reference, StandardCharsets.UTF_8));
         }
         if (isNotBlank(cardHolderName)) {
-            queries.add(CARDHOLDER_NAME_FIELD + "=" + cardHolderName);
+            queries.add(CARDHOLDER_NAME_FIELD + "=" + URLEncoder.encode(cardHolderName, StandardCharsets.UTF_8));
         }
         if (isNotBlank(firstDigitsCardNumber)) {
-            queries.add(FIRST_DIGITS_CARD_NUMBER_FIELD + "=" + firstDigitsCardNumber);
+            queries.add(FIRST_DIGITS_CARD_NUMBER_FIELD + "=" + URLEncoder.encode(firstDigitsCardNumber, StandardCharsets.UTF_8));
         }
         if (isNotBlank(lastDigitsCardNumber)) {
-            queries.add(LAST_DIGITS_CARD_NUMBER_FIELD + "=" + lastDigitsCardNumber);
+            queries.add(LAST_DIGITS_CARD_NUMBER_FIELD + "=" + URLEncoder.encode(lastDigitsCardNumber, StandardCharsets.UTF_8));
         }
         if (isSet(paymentStates)) {
-            queries.add(PAYMENT_STATES_FIELD + "=" + paymentStates.getRawString());
+            queries.add(PAYMENT_STATES_FIELD + "=" + URLEncoder.encode(paymentStates.getRawString(), StandardCharsets.UTF_8));
```

Comment:
Again, can you manually test this still works? You can select multiple payment states on the transactions page, so if you have a few transactions in various states you should be able to select various states and check it’s returning the expected results.

Reply:
👌